dev/tests/integration/testsuite/Magento/Catalog/_files/product_virtual_rollback.php

```diff
@@ -4,22 +4,25 @@
  * See COPYING.txt for license details.
  */
 
-/** @var \Magento\Framework\Registry $registry */
-$registry = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->get(\Magento\Framework\Registry::class);
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\StateException;
+use Magento\TestFramework\Helper\Bootstrap;
+
+$registry = Bootstrap::getObjectManager()->get(\Magento\Framework\Registry::class);
 
 $registry->unregister('isSecureArea');
 $registry->register('isSecureArea', true);
 
-/** @var \Magento\Catalog\Api\ProductRepositoryInterface $productRepository */
-$productRepository = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()
-    ->create(\Magento\Catalog\Api\ProductRepositoryInterface::class);
+$productRepository = Bootstrap::getObjectManager()
+    ->create(ProductRepositoryInterface::class);
 
 try {
     $product = $productRepository->get('virtual-product', false, null, true);
     $productRepository->delete($product);
-} catch (\Magento\Framework\Exception\NoSuchEntityException $exception) {
+} catch (NoSuchEntityException $exception) {
     //Product already removed
-} catch (\Magento\Framework\Exception\StateException $exception) {
+} catch (StateException $exception) {
```

Comment:
When will we have state exception?

Reply:
In case of `\Exception` in the repository method:
https://github.com/magento/graphql-ce/blob/4a28abb62b4f879794b44c2b2230db15ddea46ba/app/code/Magento/Catalog/Model/ProductRepository.php#L649

Reply:
But we expected that just `NoSuchEntityException` could be thrown if will throw another we can't suppress it